src/vs/editor/browser/editorDom.ts

```diff
@@ -205,3 +205,84 @@ export class GlobalEditorMouseMoveMonitor extends Disposable {
 		});
 	}
 }
+
+
+if (typeof ShadowRoot.prototype.caretRangeFromPoint === 'undefined') {
+
+	ShadowRoot.prototype.caretRangeFromPoint = function (x, y) {
```

Comment:
The editor should not add methods to the DOM elements. i.e. if someone embeds the editor on their page, there should be no such side-effects.

Reply:
I have reworked this to avoid changing the ShadowRoot prototype.